plugins/csp-wms-overlays/src/utils.hpp

```diff
@@ -0,0 +1,179 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+//      and may be used under the terms of the MIT license. See the LICENSE file for details.     //
+//                        Copyright: (c) 2019 German Aerospace Center (DLR)                       //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef CSP_WMS_UTILS_HPP
+#define CSP_WMS_UTILS_HPP
+
+#include "../../../src/cs-utils/convert.hpp"
+
+#include <VistaTools/tinyXML/tinyxml.h>
+
+#include <optional>
+#include <regex>
+
+namespace csp::wmsoverlays {
+
+/// Struct for storing a geographical bounding box for a map.
+/// Coordinates should be given in degrees.
+struct Bounds {
+  double mMinLon;
+  double mMaxLon;
+  double mMinLat;
+  double mMaxLat;
+
+  Bounds()
+      : mMinLon(-180.)
+      , mMaxLon(180.)
+      , mMinLat(-90.)
+      , mMaxLat(90.) {
+  }
+
+  Bounds(double minLon, double maxLon, double minLat, double maxLat)
+      : mMinLon(minLon)
+      , mMaxLon(maxLon)
+      , mMinLat(minLat)
+      , mMaxLat(maxLat) {
+  }
+
+  inline bool operator!=(const Bounds& rhs) const {
+    return mMinLon != rhs.mMinLon || mMaxLon != rhs.mMaxLon || mMinLat != rhs.mMinLat ||
+           mMaxLat != rhs.mMaxLat;
+  }
+
+  inline bool operator==(const Bounds& rhs) const {
+    return mMinLon == rhs.mMinLon && mMaxLon == rhs.mMaxLon && mMinLat == rhs.mMinLat &&
+           mMaxLat == rhs.mMaxLat;
+  }
+};
+
+/// Struct for the duration of the WMS time step.
+///	Ideally only one of the members should be non-zero.
+struct Duration {
+  int                              mYears        = 0;
+  int                              mMonths       = 0;
+  boost::posix_time::time_duration mTimeDuration = boost::posix_time::seconds(0);
+
+  bool isDuration() const;
+
+  inline bool operator==(const Duration& rhs) const {
+    return mYears == rhs.mYears && mMonths == rhs.mMonths && mTimeDuration == rhs.mTimeDuration;
+  }
+};
+
+/// Struct of timeintervals of the data set.
+struct TimeInterval {
+  boost::posix_time::ptime mStartTime;      ///< The beginning of the interval.
+  boost::posix_time::ptime mEndTime;        ///< The end of the interval.
+  std::string              mFormat;         ///< The string format of time values.
+  Duration                 mSampleDuration; ///< The duration of one sample in WMS interval.
+
+  inline bool operator==(const TimeInterval& rhs) const {
+    return mStartTime == rhs.mStartTime && mEndTime == rhs.mEndTime && mFormat == rhs.mFormat &&
+           mSampleDuration == rhs.mSampleDuration;
+  }
+};
+
+namespace utils {
+
```

Comment:
I would really appreciate a high-level comment at this point. Which time formats are used below? Why is this different than the stuff in cs::utils::convert::time?

Reply:
Looks great! Thank you so much for your work on this!